internal/files/api/syncFlags.go

```diff
@@ -52,29 +65,120 @@ func SendFlagsToAPI(results []model.FileSearchResult, envId string) (err error)
 	return err
 }
 
+func generateAuthenticationToken() error {
+	if os.Getenv("FLAGSHIP_CLIENT_ID") == "" {
+		log.Fatal("Missing required environment variable FLAGSHIP_CLIENT_ID")
+	}
+
+	if os.Getenv("FLAGSHIP_CLIENT_SECRET") == "" {
+		log.Fatal("Missing required environment variable FLAGSHIP_CLIENT_SECRET")
+	}
+
+	if os.Getenv("ACCOUNT_ID") == "" {
+		log.Fatal("Missing required environment variable ACCOUNT_ID")
+	}
+
+	if os.Getenv("FS_API") == "" {
+		log.Fatal("Missing required environment variable FS_API")
+	}
+
+	authRequest := AuthRequest{
+		GrantType:    "client_credentials",
+		Scope:        "flag_analytics.list flag_analytics.update env:*",
+		ClientId:     os.Getenv("FLAGSHIP_CLIENT_ID"),
+		ClientSecret: os.Getenv("FLAGSHIP_CLIENT_SECRET"),
+	}
+
+	body, _ := json.Marshal(authRequest)
+
+	route := fmt.Sprintf("%s/%s/token?expires_in=0", os.Getenv("FS_AUTH_API"), os.Getenv("ACCOUNT_ID"))
```

Comment:
Here why you made FS_AUTH_API an env variable instead of setting it as constant

Reply:
To be able to override it when on staging env. (FYI:`FS_API` is also an env variable ;) )

Reply:
ah yes you are right